engine/table/src/main/java/io/deephaven/engine/table/impl/locations/impl/AbstractTableLocationProvider.java

```diff
@@ -196,6 +204,16 @@ public TableLocation getTableLocationIfPresent(@NotNull final TableLocationKey t
         }
     }
 
+    private void verifyPartitionKeys(@NotNull TableLocationKey locationKey) {
+        if (partitionKeys == null) {
+            partitionKeys = new ArrayList<>(locationKey.getPartitionKeys());
+        } else if (!equals(partitionKeys, locationKey.getPartitionKeys())) {
```

Comment:
This is an improvement. We should consider also verifying the types of the partition values. It's a little bit tricky, though, since we'd want to allow for assignable types based on the schema (either inferred or supplied).